moveit_ros/moveit_servo/src/servo_calcs.cpp

```diff
@@ -618,9 +632,20 @@ bool ServoCalcs::internalServoUpdate(Eigen::ArrayXd& delta_theta,
                         servo_params_.override_velocity_scaling_factor);
 
   // Enforce SRDF position limits, might halt if needed, set prev_vel to 0
-  const auto joints_to_halt = enforcePositionLimits(next_joint_state_);
+  const auto joints_to_halt =
+      enforcePositionLimits(next_joint_state_, servo_params_.joint_limit_margin, joint_model_group_);
+
   if (!joints_to_halt.empty())
   {
+    std::ostringstream joint_names;
+    std::transform(joints_to_halt.cbegin(), std::prev(joints_to_halt.cend()),
+                   std::ostream_iterator<std::string>(joint_names, ""),
+                   [](const auto& joint) { return " '" + joint->getName() + "'"; });
+
+    joint_names << joints_to_halt.back()->getName();
```

Comment:
I missed this, but why is this line needed? Wouldn't the line above address the full set of joints if you remove the `std::prev()` ?

Here is a godbolt link showing your implementation and then the fix: https://godbolt.org/z/5hd68YGTa

Would strongly recommend testing your code before making changes. In the long run, it's easier than going back and forth with reviews.

Reply:
oh ! Really sorry about that.
Did not give it much thought since the code was from the original implementation and assumed it would work.
I wonder why it was done like this (could be bug that went unnoticed till now)

Reply:
I think the previous code worked when there were no single quotes around, and spaces after the joint name.

But because there was a `std::transform()` and another line for the last element without the trailing space, the quotes/spaces had to be updated consistently.

Anyways, after fixing that we should be good.

Reply:
Some observations while testing

If I first jog the 7th joint to reach the limit, the message correctly prints 
```
Joints 'panda_joint7' close to a position limit. Halting.
```
After this if I jog the 6th joint to the limit, the message only tells about the 6th joint while the 7th is also actually at the limit

```
Joints 'panda_joint6' close to a position limit. Halting.
```

But if I move the robot with the 6th and 7th joints at limit, using the Twist command, it gives the message as expected
```
Joints 'panda_joint6' 'panda_joint7' close to a position limit. Halting.
```

Is this the expected behaviour ? shouldn't the JointJog also print the message similar to how it works for Twsit ?
The reason I think is this check, where it skips the other joints when  the joint velocity is set as zero during jointJog for all joints except the one being moved

Reply:
That seems fine. Makes sense we don't care to check joints we're not moving.